### PR TITLE
knit: init at 1.1.1

### PR DESCRIPTION
**Pull request by DrSensor**
###### Description of changes

Add [knit](https://github.com/zyedidia/knit/tree/master#features), a build tool inspired from from GNU Make and Plane9 Mk. It use modified Lua instead of Makefile syntax.
([blog post](https://zyedidia.github.io/blog/posts/3-knit-better-make/))

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/build-managers/knit/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+buildGoModule rec {
+    pname = "knit";
+    version = "1.1.1";
+
+    src = fetchFromGitHub {
+      owner = "zyedidia";
+      repo = "knit";
+      rev = "v1.1.1";
+      sha256 = "sha256-zxwEJnQZpOEJhV7jx2ClS3XmMfGBiq8AHR26TOIBJVw=";
```
Comment (AndersonTorres) [inline pkgs/development/tools/build-managers/knit/default.nix:14]:
```suggestion
      hash = "sha256-zxwEJnQZpOEJhV7jx2ClS3XmMfGBiq8AHR26TOIBJVw=";
```

File: pkgs/development/tools/build-managers/knit/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+}:
+
+buildGoModule rec {
+    pname = "knit";
+    version = "1.1.1";
+
+    src = fetchFromGitHub {
+      owner = "zyedidia";
+      repo = "knit";
+      rev = "v1.1.1";
```
Comment (AndersonTorres) [inline pkgs/development/tools/build-managers/knit/default.nix:13]:
```suggestion
      rev = "v${version}";
```

File: pkgs/development/tools/build-managers/knit/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      sha256 = "sha256-zxwEJnQZpOEJhV7jx2ClS3XmMfGBiq8AHR26TOIBJVw=";
+    };
+
+    vendorSha256 = "sha256-+IZFydwchHIMIvYmIgZ0uJKjW4aVBFuj3SQk58I0z/g=";
```
Comment (AndersonTorres) [inline pkgs/development/tools/build-managers/knit/default.nix:17]:
```suggestion
    vendorHash = "sha256-+IZFydwchHIMIvYmIgZ0uJKjW4aVBFuj3SQk58I0z/g=";
```

File: pkgs/development/tools/build-managers/knit/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    subPackages = [
+      "cmd/knit"
+    ];
+
+    ldflags = [
+      "-s -w -X github.com/zyedidia/knit/info.Version=${version}"
```
Comment (AndersonTorres) [inline pkgs/development/tools/build-managers/knit/default.nix:24]:
```suggestion
      "-s"
      "-w"
      "-X github.com/zyedidia/knit/info.Version=${version}"
```

File: pkgs/development/tools/build-managers/knit/default.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
+
+buildGoModule rec {
+    pname = "knit";
```
Comment (AndersonTorres) [inline pkgs/development/tools/build-managers/knit/default.nix:7]: Use only two spaces as indentation.
```suggestion
  pname = "knit";
```

File: maintainers/maintainer-list.nix [hunk 6]
```diff
@@ -4457,6 +4457,13 @@
       fingerprint = "7E38 89D9 B1A8 B381 C8DE  A15F 95EB 6DFF 26D1 CEB0";
     }];
   };
+  DrSensor = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:4460]: Put this in a separate commit.

Also, squash all fixups.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by AndersonTorres** [comment]
Fix the indentation in all the file, not only on its first line.

File: maintainers/maintainer-list.nix [hunk 7]
```diff
@@ -15727,13 +15734,6 @@
     githubId = 219362;
     name = "Sarah Brofeldt";
   };
-  srid = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:15730]: Why??

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 7 above)
Comment (DrSensor) [reply maintainers/maintainer-list.nix:15730]: oops, it slip because I use `git reset master --soft`
gonna fix it

**Comment by drupol** [comment]
Could you please with the broken eval?

**Review by drupol — APPROVED** [review]
